x-pack/plugin/core/template-resources/src/main/resources/traces@settings.json

```
@@ -2,6 +2,9 @@
   "template": {
     "settings": {
       "index": {
+        "lifecycle": {
+          "name": "traces@lifecycle"
```

Totally nit, but is `@lifecycle` needed here? `logs@settings.json` and `metrics@tsd-settings.json`/`metrics@settings.json` don't have that. Asking purely for consistency.

If we go with that, then the new test in `20_traces_tests.yml` needs to be adapted as well.

```suggestion
          "name": "traces"
```

---

The non `@lifecycle` variants for `logs` and `metrics` are deprecated. However, switching to the new non-deprecated names could be considered a breaking change as users may have modified these managed ilm policies for customization (which we advise against) instead of creating a new custom ilm policy.

You're right that this creates some inconsistency, but I didn't want to introduce a new and immediately deprecated `traces` ilm policy.

Another option would be to declare the attempt to rename the ilm policies to `...@lifecycle` a failed attempt and remove the deprecation warning for the non `@lifecycle` variants.